### PR TITLE
MANKIND IS DEAD. BLOOD IS FUEL. HELL IS FULL. IPC MARTIAL ART

### PR DESCRIPTION
**Pull request by Moltijoe**
Gives hijack and martyr IPC traitors a unique martial art.
This is a super blatant Ultrakill reference if you hadn't guessed already, though someone a bit more discrete can definitely hide it better than me.
It's mainly focused on moving fast and causing as much bloodshed as possible
I'm absolute shit at balance, these are just random numbers for now, mainly to test.

### **Features:**
Replaces disarm with a dash that has charges and a cooldown
Removes your ability to grab (mainly because grabbing slows down combat)
Lets you "redirect" emps using throwmode
Gives temporary emp immunity after countering or getting hit by an emp
Completing any combo will give a speed buff with a duration scaling based on combo difficulty
Getting covered in blood will heal

### **Combos:**
Grab-Grab (Pocket Pistol) puts a single shot revolver in your hand (has to be a living person, or yourself)
Grab-Harm-Grab (Gun Hand) puts a single round shotgun in your hand (has to be a living person not in crit)
Harm-Harm-Harm (Blood Burst) Causes the enemy to lose some blood, shooting some out around them and covering you in blood. 
Blood burst executes people in hardcrit shooting significantly more blood and healing

Charge counter

:cl:  
rscadd: Added a new hijack and glorious death only ipc martial art
imageadd: added dash charge icon
/:cl:

### CODE REVIEW

**Comment by VaelophisNyx** [comment]
you GOTTA make it come with the wings potion, come on

**Comment by Moltijoe** [comment]
> you GOTTA make it come with the wings potion, come on

**Comment by Moltijoe** [comment]
Oopsie, meant to say, sure, why not, looks cool and isn't that strong

**Comment by GraveHat** [comment]
Fast-paced martial art without grabs!? Sounds like its time to dodge ions.

File: code/datums/martial.dm [hunk 1]
```diff
@@ -3,7 +3,7 @@
   * Martial Arts
   *
   * Martial arts change how human unarmed attacks effect other humans, and allow for strings of attacks on the same target to have special effects
-  * Martial arts also can block melee or ranged attacks using block_chance and deflect_chance variables
+  *   arts also can block melee or ranged attacks using block_chance and deflect_chance variables
```
Comment (VaelophisNyx) [inline code/datums/martial.dm]: Why tho

Comment (Moltijoe) [reply code/datums/martial.dm]: uh, yeah, totally intended to do that, and i'm not just a bull in a china shop with all this code

**Comment by Moltijoe** [comment]
> Fast-paced martial art without grabs!? Sounds like its time to dodge ions.

currently i've given them stunimmune to prevent iron+uranium from just ending them instantly, it still deals tons of damage, including brain damage though.
shouldn't be a problem because if they're using this, trying to stun them is just the boring way out.
if that's not enough to prevent a single amateur chemist from deleting them i might have to do some other things

**Comment by Moltijoe** [comment]
the counter effect of emps needs to be strong enough to dissuade their use without invalidating them.

**Comment by Moltijoe** [comment]
yeah, i'm also giving them "adaptive circuits" if they get hit by, or counter an emp it'll give a short lived immunity to it

**Comment by Moltijoe** [comment]
Not entirely certain how to fix the error, i'm sure someone more experienced does know though

**Comment by Marmio64** [comment]
this doesnt have the ability to swat projectiles away like carp, yes? Stun immunity + no damage slowdown + self sustainability + projectile 'immunity' + still being able to use guns would be far too powerful.

**Comment by Moltijoe** [comment]
> this doesnt have the ability to swat projectiles away like carp, yes? Stun immunity + no damage slowdown + self sustainability + projectile 'immunity' + still being able to use guns would be far too powerful.

Nope, no bullet reflection, only emp reflection

**Review by JamieD1 — APPROVED** [review]

**Comment by JamieD1** [comment]
Cornflakes

**Comment by Moltijoe** [comment]
> Cornflakes

what?

**Comment by UltraBigRat** [comment]
why is this hijack only though? the abilities dont really seem to be only murderbone focused you could use this well without having to kill everyone. really restricts this to be extremely rare especially with it costing 20 tc it shouldnt be too bad

**Comment by Moltijoe** [comment]
> why is this hijack only though? the abilities dont really seem to be only murderbone focused you could use this well without having to kill everyone. really restricts this to be extremely rare especially with it costing 20 tc it shouldnt be too bad

you're kidding, right? one of the comboes is literally an execute that heals, that incentivizes finishing people off for the heal.

**Comment by UltraBigRat** [comment]
> > why is this hijack only though? the abilities dont really seem to be only murderbone focused you could use this well without having to kill everyone. really restricts this to be extremely rare especially with it costing 20 tc it shouldnt be too bad
> 
> you're kidding, right? one of the comboes is literally an execute that heals, that incentivizes finishing people off for the heal.

Im not really kidding? one ability being focuse on finish people off in exchange for heals does not mean you are required to immediatly murderbone everyone, can if anything just be used as a good way to finish off targets. I dont know how good my opinion here is but, i just feel having a single ability that rewards murder doesnt equal murderbone only item. Kind of like saying bloodsuckers must go around murderboning everyone because theyre rewarded for killing people with blood.

**Review by cuackles — APPROVED** [review]

**Comment by Ziiro** [comment]
I'm very late to the party on this but I wanted to comment that this kicks ass in your translation of the concepts to SS13, good stuff

**Pull request merged**